frontend/js/angular/controller.js

```diff
@@ -706,48 +709,61 @@ RodiApp.controller('RodiCtrl', ['$scope', 'RodiSrv', '$window', '$filter', '$loc
 
-                        var lastScoreValue = $scope.countriesList[$scope.countriesList.length - 1].score * 1;
+                        var lastScoreValue = $scope.countriesListWithScore[$scope.countriesListWithScore.length - 1].score * 1;
                         var rankValue = 0;
 
                         if(lastScoreValue == 0){
-                            rankValue = $scope.countriesList[$scope.countriesList.length - 1].rank;
+                            rankValue = $scope.countriesListWithScore[$scope.countriesListWithScore.length - 1].rank;
                         } else {
-                            rankValue = ($scope.countriesList[$scope.countriesList.length - 1].rank * 1) + 1;
+                            rankValue = ($scope.countriesListWithScoret[$scope.countriesListWithScore.length - 1].rank * 1) + 1;
                         }
 
-                        console.log(lastScoreValue);
-
-                        for (i=0; i< $scope.allCountries.length; i++)
+                        for (i = 0; i < $scope.allCountries.length; i++)
                         {
-                            // For esch country create e rank item if no dataset
-                            filterResult = $filter('filter')($scope.countriesList, function(item)
+                            // For all countries build the structure for table
+                            // Search if this country has a score set
+                            filterResult = $filter('filter')($scope.countriesListWithScore, function(item)
                             {
                                 return item.country == $scope.allCountries[i].iso2;
                             });
+
+                            // This country doesn't have a score - without dataset submitted
                             if(filterResult.length == 0){
-                            //     Add country in the obj Array
+                                //     Add country in the obj Array
                                 $scope.objCountry = {
                                     country: $scope.allCountries[i].iso2,
+                                    description: $scope.allCountries[i].name,
                                     datasets_count: 0,
                                     fullscores_count: 0,
                                     rank: rankValue,
                                     score: 0
                                 }
 
+                                // Add country in the structure for table
                                 $scope.countriesList.push($scope.objCountry);
 
-                                // $scope.countriesTable = new NgTableParams({
-                                //     page: 1,
-                                //     count: 20
-                                // }, {
-                                //     total: $scope.countriesList.length
-                                // });
+                            } else {
```

Comment:
@CIMAManuel I don't understand why you duplicate the object whereas it is supposed to be a sort function?

Reply:
@oncleton I don't duplicate items. I have two API to call before create the table list.
One API give me all countries available and the second return to me a countries list with ranking.
I must to merge "$scope.allCountries" into "$scope.countriesList" to have datasets with ranking and dataset without ranking in the same table.
So if the country has not rank I insert the first object else I can add a object with more data.
This stuff it's do the first time in initPage function. The sort function in "orderArrayByProperty" where i do only e orderBy filter.